src/okareo/model_under_test.py

```diff
@@ -451,6 +469,108 @@ def _extract_input_from_scenario_data_point(
+        if self.models.get("custom"):
+            return self.models["custom"]["model_invoker"](args)
+        elif self.models.get("custom_batch"):
+            return self.models["custom_batch"]["model_invoker"](args)
+        else:
+            return self.models["driver"]["target_params"]["model_invoker"](args)
+
+    def get_params_from_custom_result(self, result: Any) -> Any:
+        if isinstance(result, ModelInvocation):
+            result = result.params()
+        if (
+            isinstance(result, list)
+            and len(result) > 0
+            and result[0].get("model_invocation")
+        ):
+            result = [r["model_invocation"].params() for r in result]
+        return result
+
+    async def run_async_tasks(self, stop_event: Any, nats_jwt: str, seed: str) -> None:
+        nats_connection = await self.connect_nats(nats_jwt, seed)
+        try:
+
+            async def message_handler_custom_model(msg: Any) -> None:
+                try:
+                    data = json.loads(msg.data.decode())
+                    args = data.get("args", [])
+                    result = self.call_custom_invoker(args)
+                    json_encodable_result = self.get_params_from_custom_result(result)
+                    await nats_connection.publish(
+                        msg.reply, json.dumps(json_encodable_result).encode()
+                    )
+                except Exception as e:
+                    print(f"An error occurred in the custom model invocation: {str(e)}")
+
+            await nats_connection.subscribe(
+                f"invoke.{self.mut_id}", cb=message_handler_custom_model
+            )
+            await stop_event.wait()
+        except Exception as e:
+            print(f"An error occurred in the custom model invocation: {str(e)}")
+        finally:
+            await nats_connection.close()
+
+    def run_async_in_thread(self, coro: Any) -> Any:
+        loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(loop)
+        result = loop.run_until_complete(coro)
+        return result
+
+    def start_custom_model_thread(self, nats_jwt: str, seed: str) -> tuple:
+        custom_model_thread_stop_event = asyncio.Event()
+        custom_model_thread = threading.Thread(
+            target=self.run_async_in_thread,
+            args=(
+                self.run_async_tasks(custom_model_thread_stop_event, nats_jwt, seed),
+            ),
+        )
+        return custom_model_thread, custom_model_thread_stop_event
+
+    def cleanup_custom_model(
+        self, custom_model_thread_stop_event: Any, custom_model_thread: Any
+    ) -> None:
+        if custom_model_thread_stop_event:
+            custom_model_thread_stop_event.set()  # Signal the async task to stop
+            custom_model_thread_stop_event = None
+
+        if custom_model_thread:
+            custom_model_thread.join(timeout=1)
+            custom_model_thread = None
+
+        remove_nats_user_v0_remove_nats_user_delete.sync(
```

Comment:
I was having trouble cleaning up the connection, so I went with this method. This was this issue:
Client
1 thread has connection
1 thread has run_test request

If we delete the user on the server at the end of the run_test call, this would abruptly terminate the connection on the SDK, and the nats python client would print out an error on the client. To avoid this, I had the client call the delete user method.

I see another path though, and this works:
At the end of the run_test call, I will send a "nats close" message to to the SDK, which will then close the connection and then when we delete the user on the server, there is no abrupt termination